docs/source/pjsua2/using/media_video.rst

```diff
@@ -0,0 +1,301 @@
+non-GUI thread too, for example we found :cpp:func:`pj::Endpoint::libDestroy()` in C# desktop
+will cause stuck when initiated from GUI thread.
+
+Generally it is a good practice to keep the GUI thread free from non-UI work to improve application
+responsiveness. So it is also recommended to avoid calling PJSIP API from GUI thread since:
+
+- it may take some time to complete, or
+- it may block while trying to acquire a lock.
+    
+Here is a sample code to post a job via schedule timer, in this sample, it is for scheduling
+a video capture device preview start.
+
+.. code-block:: c++
+
+    // Timer type ID
+    enum {
+        TIMER_START_PREVIEW = 1,
+	...
+    };
+    
+    // Generic timer parameter
+    struct MyTimerParam {
+	int type;
+	union {
+	    struct {
+	        int   dev_id;
+		void *hwnd;
+		int   w, h, fps;
+	    } start_preview;
+	    ...
+	} data;
+    };
+    
+    
+    // PJSUA2 Endpoint::onTimer() implementation
+    void Endpoint::onTimer(const OnTimerParam &prm)
+    {
+        MyTimerParam *param = (MyTimerParam*) prm.userData;
+	if (param->type == TIMER_START_PREVIEW) {
+	    int dev_id = param->data.start_preview.dev_id;
+            void *hwnd = param->data.start_preview.hwnd;
+	    int w      = param->data.start_preview.w;
+	    int h      = param->data.start_preview.h;
+	    int fps    = param->data.start_preview.fps;
+	    StartPreview(device_id, hwnd, w, h, fps);
+	}
+	...
+	
+	// Finally delete the timer parameter.
+	delete param;
+    }
+    
+    ...
+    
+    MyTimerParam *tp = new MyTimerParam();
+    tp->type = TIMER_START_PREVIEW;
+    tp->data.start_preview.dev_id = 1; // colorbar virtual device
+    tp->data.start_preview.hwnd   = (void*)some_hwnd;
+    tp->data.start_preview.w      = 320;
+    tp->data.start_preview.h      = 240;
+    tp->data.start_preview.fps    = 15;
+    
+    // Schedule the preview start to be executed immediately (zero milisecond delay).
+    Endpoint::instance().utilTimerSchedule(0, tp);
+
+
+Call's video media
+----------------------------
+Unlike in audio, call video media is separated between encoding and decoding, this is because
+the video formats (e.g: width, height, frame rate) of both directions can be different.
+Application can retrieve the video media objects using 
+:cpp:func:`pj::Call::getEncodingVideoMedia()` for the encoding direction and
+:cpp:func:`pj::Call::getDecodingVideoMedia()` for the decoding direction, both will return
+instance of class :cpp:class:`pj::VideoMedia`.
+    
+Also unlike in audio call where port connections between audio device and call audio media needs
+to be set up manually by application, in video, the port connections in the conference bridge
+are set up automatically by the library, so the video capture device (configured via account settings)
+will be connected to the encoding video media and the decoding video media will be connected
+to a renderer video window.
```

Comment:
maybe add this at the end of the sentence to clarify:  "once the call's video media is started".